### ISSUE TITLE
[QUESTION] NPM using https instead of ssh

### ISSUE THREAD

**Issue by michalhlavka** [npm/cli#1396]
Hi, 

in dependencies, I've got 
`"package": "git+ssh://git@bitbucket.org:namespace/package.git"` and NPM always try to pull from HTTPS. I was trying a specific version of the repository, specific branch, no chance to set it by ssh. 

```
npm ERR! /usr/bin/git ls-remote -h -t https://bitbucket.org/namespace/package.git
npm ERR! 
npm ERR! remote: Invalid username or password
npm ERR! fatal: Authentication failed for 'https://bitbucket.org/namespace/package.git/'
npm ERR! 
npm ERR! exited with error code: 128
```

Only once it was by ssh, but this happened when I set the wrong ssh key. 

```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@bitbucket.org/namespace/package.git
npm ERR! 
npm ERR! Warning: Permanently added the RSA host key for IP address '18.205.93.2' to the list of known hosts.
npm ERR! git@bitbucket.org: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
```

NPM version 6.14.4

I will be glad for any help,
thank you.

**Comment by darcyclarke** [comment]
npm `v6` is no longer in active development; We will continue to push security releases to `v6` at our team's discretion as-per our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy).

If your bug is preproducible on `v7`, please re-file this issue using our [new issue template](https://github.com/npm/cli/issues/new?assignees=&labels=Bug%2CNeeds+Triage%2CRelease+7.x&template=bug_7.yml&title=%5BBUG%5D+%3Ctitle%3E).

If your issue was a feature request, please consider opening a new [RRFC](https://github.com/npm/rfcs/issues) or [RFC](https://github.com/npm/rfcs). If your issue was a question or other idea that was not CLI-specific, consider opening a discussion on our [feedback repo](https://github.com/npm/feedback)

**Closing:** This is an automated message.
